src/settings/rageshake.ts

```diff
@@ -550,12 +553,21 @@ type StringifyReplacer = (
 // Injects `<$ cycle-trimmed $>` wherever it cuts a cyclical object relationship
 const getCircularReplacer = (): StringifyReplacer => {
   const seen = new WeakSet();
-  return (key: string, value: unknown): unknown => {
+  const depthMap = new WeakMap<object, number>();
+  return function (this: unknown, key: string, value: unknown): unknown {
     if (typeof value === "object" && value !== null) {
       if (seen.has(value)) {
         return "<$ cycle-trimmed $>";
       }
       seen.add(value);
+
+      let depth = 0;
+      if (this) {
+        depth = depthMap.get(this) ?? 0;
+      }
```

Comment:
```suggestion
      let depth = depthMap.get(this) ?? 0;
```
since the keys need to be object values this should be enough? `get(undefined)` is always undefined I think.

Reply:
> get(undefined) is always undefined I think.

```
> const x = new Map();
> x.set(undefined, "hello");
Map(1) {undefined => 'hello'}
> x.get(undefined);
'hello'
```

So... nope. Arguably since we only enter this block if value is an object, we should never add `undefined` as a key, but I would have to reason through all of that before understanding the code. To me, I think it would be a lot more readable with the explicit `if` statement since it follows the logic: if the object has a parent, look up its parent's depth in the map. If it has no parent, it's depth is zero.

Reply:
With the WeekMap we are using it is not possible however?
```
> const depthMap = new WeakMap();
> undefined
> depthMap.set(undefined, 10)
> Uncaught TypeError: WeakMap key must be an object, got undefined
    <anonymous> debugger eval code:1
 ```
 But I see your point that this is clearer! Thanks for clarifying (looking into it again)